provisioning/Samples/device/SymmetricKeySample/ProvisioningDeviceClientSample.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Provisioning.Client.Transport;
+using Microsoft.Azure.Devices.Shared;
+using System;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Provisioning.Client.Samples
+{
+    /// <summary>
+    /// Demonstrates how to register a device with the device provisioning service, and then
+    /// use the registration information to authenticate to IoT Hub.
+    /// </summary>
+    internal class ProvisioningDeviceClientSample
+    {
+        private readonly Parameters _parameters;
+
+        public ProvisioningDeviceClientSample(Parameters parameters)
+        {
+            _parameters = parameters;
+        }
+
+        public async Task RunSampleAsync()
+        {
+            if (_parameters.EnrollmentType == EnrollmentType.Group)
+            {
+                _parameters.PrimaryKey = ComputeDerivedSymmetricKey(_parameters.PrimaryKey, _parameters.RegistrationId);
+                _parameters.SecondaryKey = ComputeDerivedSymmetricKey(_parameters.SecondaryKey, _parameters.RegistrationId);
+            }
+
+            Console.WriteLine($"Initializing the device provisioning client...");
+            ProvisioningTransportHandler transportHandler = GetTransportHandler();
+
+            using var security = new SecurityProviderSymmetricKey(
+                _parameters.RegistrationId,
+                _parameters.PrimaryKey,
+                _parameters.SecondaryKey);
+            ProvisioningDeviceClient provClient = ProvisioningDeviceClient.Create(
+                _parameters.GlobalDeviceEndpoint,
+                _parameters.IdScope,
+                security,
+                transportHandler);
+
+            Console.WriteLine($"Initialized for registration Id {security.GetRegistrationID()}");
+
+            Console.WriteLine("Registering with the device provisioning service... ");
+            DeviceRegistrationResult result = await provClient.RegisterAsync().ConfigureAwait(false);
+
+            Console.WriteLine($"Registration status: {result.Status}.");
+            if (result.Status != ProvisioningRegistrationStatusType.Assigned)
+            {
+                return;
```

Comment:
We return here without telling a user why we stopped the sample. Do we need to print a console message to the user on what the actual status is before returning?